scripts/abi_check.py

```diff
@@ -114,11 +118,24 @@ def update_git_submodules(self, git_worktree_path):
         self.log.info(output.decode("utf-8"))
         if process.returncode != 0:
             raise Exception("git submodule update failed, aborting")
+        if (os.path.exists(os.path.join(git_worktree_path, "crypto"))
+                and crypto_rev):
+            checkout_process = subprocess.Popen(
+                [self.git_command, "checkout", crypto_rev],
+                cwd=os.path.join(git_worktree_path, "crypto"),
+                stdout=subprocess.PIPE,
+                stderr=subprocess.STDOUT
+            )
+            checkout_output, _ = checkout_process.communicate()
+            self.log.info(checkout_output.decode("utf-8"))
+            if checkout_process.returncode != 0:
+                raise Exception("git checkout failed, aborting")
 
     def build_shared_libraries(self, git_worktree_path):
         my_environment = os.environ.copy()
         my_environment["CFLAGS"] = "-g -Og"
         my_environment["SHARED"] = "1"
+        my_environment["USE_CRYPTO_SUBMODULE"] = "1"
```

Comment:
Is this going to always build with the submodule for versions of TLS that supporting building with submodule? This is a change worth mentioning in the commit message.

Reply:
As it is, it will always try to use the submodule. Do you think its worth adding options to enable testing with and without the submodule?

Reply:
Always using the submodule makes sense, since we want to test Crypto mainly (and TLS will only be able to use the submodule after `mbedtls_*` crypto functions move there) as does falling back to not use the submodule if it isn't available.

Reply:
Please add a note in the commit message that introduces this change.

Reply:
Updated the commit message.